File: mycollab-services-ondemand/src/main/java/com/mycollab/ondemand/module/file/service/AmazonServiceConfiguration.java
package com.mycollab.ondemand.module.file.service;

import com.amazonaws.services.s3.AmazonS3;
import com.amazonaws.services.s3.AmazonS3ClientBuilder;
import com.mycollab.configuration.ApplicationProperties;

import java.util.Properties;

/**
 * @author MyCollab Ltd
 * @since 5.3.0
 */
public final class AmazonServiceConfiguration {
    private static final String AWS_KEY = "s3.key";
    private static final String AWS_SECRET_KEY = "s3.secretKey";
    private static final String BUCKET = "s3.bucket";

    private String bucket;

    private static final AmazonServiceConfiguration instance = new AmazonServiceConfiguration();

    private void AmazonServiceConfiguration() throws Exception {
        Properties props = ApplicationProperties.getAppProperties();
        String awsKey = props.getProperty(AWS_KEY);
        String awsSecretKey = props.getProperty(AWS_SECRET_KEY);

        bucket = props.getProperty(BUCKET);

        if ("".equals(awsKey) || "".equals(awsSecretKey) || "".equals(bucket)) {
            throw new IllegalArgumentException(
                    "Invalid s3 configuration. All values awsKey, awsSecretKey, bucket must be set");
        }
        System.setProperty("aws.accessKeyId", awsKey);
        System.setProperty("aws.secretKey", awsSecretKey);
    }

    public static AmazonServiceConfiguration getInstance() {
        return instance;
    }

    public AmazonS3 newS3Client() {
        return AmazonS3ClientBuilder.defaultClient();
    }

    public String getBucket() {
        return bucket;
    }
}

File: mycollab-services-ondemand/src/main/java/com/mycollab/ondemand/module/file/service/AmazonServiceConfiguration.kt
package com.mycollab.ondemand.module.file.service

import com.amazonaws.SDKGlobalConfiguration
import com.amazonaws.services.s3.AmazonS3
import com.amazonaws.services.s3.AmazonS3ClientBuilder
import com.mycollab.spring.AppContextUtil
import org.slf4j.LoggerFactory

/**
 * @author MyCollab Ltd
 * @since 5.3.0
 */
class AmazonServiceConfiguration @Throws(Exception::class) private constructor() {

    var bucket: String? = null
        private set

    init {
        val s3Configuration = AppContextUtil.getSpringBean(S3Configuration::class.java)
        val awsKey = s3Configuration.key
        val awsSecretKey = s3Configuration.secretKey
        bucket = s3Configuration.bucket
        if ("" == awsKey || "" == awsSecretKey || "" == bucket) {
            throw IllegalArgumentException(
                    "Invalid s3 configuration. All values awsKey, awsSecretKey, bucket must be set")
        }
        System.setProperty(SDKGlobalConfiguration.ACCESS_KEY_SYSTEM_PROPERTY, awsKey)
        System.setProperty(SDKGlobalConfiguration.SECRET_KEY_SYSTEM_PROPERTY, awsSecretKey)
    }

    fun newS3Client(): AmazonS3 = AmazonS3ClientBuilder.defaultClient()

    companion object {
        val LOG = LoggerFactory.getLogger(AmazonServiceConfiguration::class.java)

        @JvmStatic
        val instance = AmazonServiceConfiguration()
    }
}
